packages/core/test/actions.test.ts

```diff
@@ -2603,6 +2604,103 @@ describe('sendTo', () => {
 
     interpret(parentMachine).start();
   });
+
+  it('should be able to read from event', () => {
+    expect.assertions(1);
+    const machine = createMachine<any, any>({
+      initial: 'a',
+      context: () => ({
+        foo: spawn((_, receive) => {
+          receive((event) => {
+            expect(event).toEqual({ type: 'EVENT' });
+          });
+        })
+      }),
+      states: {
+        a: {
+          on: {
+            EVENT: {
+              actions: sendTo((ctx, e) => ctx[e.value], { type: 'EVENT' })
+            }
+          }
+        }
+      }
+    });
+
+    const service = interpret(machine).start();
+
+    service.send({ type: 'EVENT', value: 'foo' });
+  });
+});
+
+describe('raise', () => {
+  it('should be able to send an event to itself', (done) => {
+    const machine = createMachine({
+      initial: 'a',
+      states: {
+        a: {
+          entry: raise(
+            { type: 'EVENT' },
+            {
+              delay: 100
+            }
+          ),
+          on: {
+            TO_B: 'b'
+          }
+        },
+        b: {
+          on: {
+            EVENT: 'c'
+          }
+        },
+        c: {
+          type: 'final'
+        }
+      }
+    });
+
+    const service = interpret(machine).start();
+
+    service.onDone(() => done());
+
+    // Ensures that the delayed self-event is sent when in the `b` state
+    service.send('TO_B');
```

Comment:
To avoid type errors when merging this into `next` we can do this:
```suggestion
    service.send({ type: 'TO_B' });
```